client/src/main/java/com/mytiki/publish/client/auth/AuthService.kt

```diff
@@ -20,69 +21,84 @@ import kotlin.io.encoding.ExperimentalEncodingApi
  */
 class AuthService {
 
+fun providerToken(): Deferred<String> = CoroutineScope(Dispatchers.IO).async {
+    val body = FormBody.Builder()
+        .add("grant_type", "client_credentials")
+        .add("client_id","provider:${TikiClient.config.providerId}")
+        .add("client_secret", TikiClient.config.publicKey)
+        .add("scope", "account:provider")
+        .add("expires", "600")
+        .build()
+    val response = ApiService.post(
+        header =  mapOf(
+            "Accept" to "application/json",
+            "Content-Type" to "application/x-www-form-urlencoded"
+        ),
+        endPoint = "https://account.mytiki.com/api/latest/auth/token",
+        onError = Exception("error on getting token"),
+        body
+    ).await()
+    TikiTokenResponse.fromJson(JSONObject(response?.string()!!)).access_token
+}
 
-    /**
-     * Retrieves the authentication token, refreshing if necessary.
-     * @param providerID The provider ID.
-     * @param publicKey The public key.
-     * @return The authentication token.
-     */
-    fun token(): CompletableDeferred<String> {
-        val token = CompletableDeferred<String>()
-        CoroutineScope(Dispatchers.IO).launch {
-            val body = FormBody.Builder()
-                .add("grant_type", "client_credentials")
-                .add("client_id", "provider:${TikiClient.config.providerId}")
-                .add("client_secret", TikiClient.config.publicKey)
-                .add("scope", "account:provider trail publish")
-                .add("expires", "600")
-                .build()
-
-            val response = ApiService.post(
-                mapOf("accept" to "application/json"),
-                "https://account.mytiki.com/api/latest/auth/token",
-                body,
-                Exception("error on getting token")
-            ).await()
-
-            token.complete(TikiTokenResponse.fromJson(JSONObject(response?.string()!!)).access_token)
-        }
-        return token
-    }
+@OptIn(ExperimentalEncodingApi::class)
```

Comment:
Remove experimental APIs. Our code must be as stable and compatible as possible, since it is a library.